includes/documents/class-wcpdf-invoice.php

```diff
@@ -103,25 +103,50 @@ public function init_number() {
 			return $invoice_number;
 		}
 
-		$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method();
-		$number_store_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', 'invoice_number', $this );
-		$number_store = new Sequential_Number_Store( $number_store_name, $number_store_method );
-		// reset invoice number yearly
-		if ( isset( $this->settings['reset_number_yearly'] ) ) {
-			$current_year = date("Y");
-			$last_number_year = $number_store->get_last_date('Y');
-			// check if we need to reset
-			if ( $current_year != $last_number_year ) {
-				$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', 1, $this ) );
-			}
+		if( ! empty( $document_date = $this->get_date() ) ) {
+			$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method();
+			$number_store_name   = $this->store_name( $document_date );
+			$number_store        = new Sequential_Number_Store( $number_store_name, $number_store_method );
+			$invoice_number      = $number_store->increment( $this->order_id, $document_date->date_i18n( 'Y-m-d H:i:s' ) );
+
+			$this->set_number( $invoice_number );
+
+			return $invoice_number;
 		}
+	}
 
-		$invoice_date = $this->get_date();
-		$invoice_number = $number_store->increment( $this->order_id, $invoice_date->date_i18n( 'Y-m-d H:i:s' ) );
+	public function store_name( $document_date ) {
+		$document_date      = date( 'Y', strtotime( $document_date ) );
+		$current_date       = date( 'Y' );
+		$reset_numeration   = isset( $this->settings['reset_number_yearly'] ) ? true : false;
+		$current_store_name = "{$this->slug}_number_{$document_date}";
 
-		$this->set_number( $invoice_number );
+		// reset: on
+		if( $reset_numeration ) {
+			$store_name          = $current_store_name;
+			$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method();
+			$number_store        = new Sequential_Number_Store( $store_name, $number_store_method );
+			$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', 1, $this ) );
```

Comment:
this will reset the number each document. Maybe you want to do this only if the store exists? You could make that check the primary condition and make the `$reset_numeration` check secondary